### PR TITLE
[stable/coredns] Add optional cluster-proportional-autoscaler

### PR DESCRIPTION
**Pull request by Smirl**
#### Is this a new chart
No

#### What this PR does / why we need it:

The coredns chart doesn't include any sort of autoscaling and default to 1 replica.
This adds a [cluster-proportional-autoscaler](https://github.com/kubernetes-incubator/cluster-proportional-autoscaler) similar to the one that `kops` creates when creating coredns. See [k8s-1.12.yaml.template](https://github.com/kubernetes/kops/blob/master/upup/models/cloudup/resources/addons/coredns.addons.k8s.io/k8s-1.12.yaml.template).

This defaults to disabled to keep backwards compatibility.

#### Which issue this PR fixes
None

#### Special notes for your reviewer:
I have copied a lot of the style from the other templates in this chart and tried to only touch the parts that needed to be changed.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/coredns/README.md [hunk 1]
```diff
@@ -74,7 +74,21 @@ The command removes all the Kubernetes components associated with the chart and
 | `tolerations`                           | Tolerations for pod assignment                                                        | []                                                          |
 | `zoneFiles`                             | Configure custom Zone files                                                           | []                                                          |
 | `extraSecrets`                          | Optional array of secrets to mount inside the CoreDNS container                       | []                                                          |
-| `customLabels`			  | Optional labels for Deployment, Pod, Service, ServiceMonitor objects		  | {}								|
+| `customLabels`                          | Optional labels for Deployment(s), Pod, Service, ServiceMonitor objects               | {}                                                          |
+| `autoscaler.enabled`                    | Optionally enabled a cluster-proportional-autoscaler for coredns                      | `false`                                                     |
```
Comment (mrueg) [inline stable/coredns/README.md]: replace coredns with CoreDNS to match upstream naming (also following)

File: stable/coredns/templates/autoscaler.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
+{{- if .Values.autoscaler.enabled }}
+---
```
Comment (mrueg) [inline stable/coredns/templates/autoscaler.yaml]: Please separate into one yaml file per object post-fixed with autoscaler e.g. `deployment-autoscaler.yaml`

File: stable/coredns/templates/autoscaler.yaml [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        {{- end }}
+      annotations:
+        {{- if .Values.isClusterService }}
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        scheduler.alpha.kubernetes.io/tolerations: '[{"key":"CriticalAddonsOnly", "operator":"Exists"}]'
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "coredns.fullname" . }}-autoscaler
+      {{- if .Values.priorityClassName }}
+      priorityClassName: {{ .Values.priorityClassName | quote }}
```
Comment (mrueg) [inline stable/coredns/templates/autoscaler.yaml]: does the autoscaler need the same priority class than coredns?

Comment (Smirl) [reply stable/coredns/templates/autoscaler.yaml]: I think it is important it does. So if the cluster is under contention we could ensure that coredns can still scale. However, I can make it a separate value which defaults to the same, if people want to lower it (or raise it)?

Comment (mrueg) [reply stable/coredns/templates/autoscaler.yaml]: Separate value with defaulting to priorityClassName sounds good.

File: stable/coredns/templates/clusterrole-autoscaler.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{{- if .Values.autoscaler.enabled }}
```
Comment (mrueg) [inline stable/coredns/templates/clusterrole-autoscaler.yaml]: This probably needs to check on .Values.rbac.create as well?

File: stable/coredns/templates/clusterrolebinding-autoscaler.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.autoscaler.enabled }}
```
Comment (mrueg) [inline stable/coredns/templates/clusterrolebinding-autoscaler.yaml]: This probably needs to check on .Values.rbac.create as well?

**Pull request merged**